infra_tooling_account/infra_tooling_account/infra_tooling_grafana_stack.py

```diff
@@ -0,0 +1,383 @@
+        These include the Timestream database ARN, Timestream KMS key ARN, settings S3 bucket and its ARN.
+
+        Returns:
+            tuple: A tuple containing references to the Timestream database ARN, Timestream KMS key ARN,
+                   settings S3 bucket and its ARN.
+        """
+
+        input_timestream_database_arn = Fn.import_value(
+            AWSNaming.CfnOutput(self, "metrics-events-storage-arn")
+        )
+        input_timestream_kms_key_arn = Fn.import_value(
+            AWSNaming.CfnOutput(self, "metrics-events-kms-key-arn")
+        )
+        input_settings_bucket_arn = Fn.import_value(
+            AWSNaming.CfnOutput(self, "settings-bucket-arn")
+        )
+
+        settings_bucket = s3.Bucket.from_bucket_arn(
+            self,
+            "salmonSettingsBucket",
+            bucket_arn=input_settings_bucket_arn,
+        )
+
+        return (
+            input_timestream_database_arn,
+            input_timestream_kms_key_arn,
+            input_settings_bucket_arn,
+            settings_bucket,
+        )
+
+    def create_grafana_admin_secret(self) -> tuple[str, str]:
+        """
+        Creates Grafana admin password in Secrets Manager.
+
+        Returns:
+            tuple: A tuple containing Grafana admin secret name and its ARN
+        """
+        grafana_admin_secret = secretsmanager.Secret(
+            self,
+            "GrafanaSecret",
+            secret_name=AWSNaming.Secret(self, "grafana-password"),
+            description="Grafana secret stored in AWS Secrets Manager",
+            generate_secret_string=secretsmanager.SecretStringGenerator(
+                include_space=False,
+                generate_string_key="password",
+                password_length=20,
+                secret_string_template='{"username":"admin"}',
+            ),
+        )
+
+        return grafana_admin_secret.secret_name, grafana_admin_secret.secret_arn
+
+    def create_grafana_key_pair(self) -> str:
+        """
+        Creates Key Pair for Grafana instance.
+
+        Returns:
+            str: Key Pair name
+        """
+        grafana_key_pair = ec2.CfnKeyPair(
+            self,
+            "GrafanaKeyPair",
+            key_name=AWSNaming.KMSKey(self, "grafana-key-pair"),
+        )
+
+        return grafana_key_pair.key_name
+
+    def prepare_and_upload_grafana_configuration_files(self, settings_bucket) -> None:
+        """
+        Generates and uploads Grafana configuration files to S3 bucket.
+        For each Service, a separate dashboard and data source will be provisioned.
+        The files will be uploaded to settings/grafana dedicated folder.
+
+        Args:
+            settings_bucket (s3.Bucket): Settings S3 Bucket
+        """
+        metric_table_names = {x: f"{x}-metrics" for x in SettingConfigs.RESOURCE_TYPES}
+        services = metric_table_names.keys()
+        timestream_database_name = AWSNaming.TimestreamDB(
```

Comment:
Can we use the output from infra_tooling_common_stack? There is an output with the DB name from it.

Reply:
unfortunately, if I use the output with the DB name in the Grafana generated config files that will be uploaded to S3 bucket, I'm getting this error:
RuntimeError: Invalid CloudFormation reference. Key must start with any of "Ref" or "Fn::GetAtt" or "Fn::Select". Got {"Fn::ImportValue":"output-salmon-metrics-events-db-name-devay"}

will try to research further on this